receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -41,33 +46,46 @@ const (
 	gcIntervalDelta   = 1 * time.Minute
 )
 
+// closeOnce ensures that the scrape manager is started after the discovery manager
+type closeOnce struct {
+	C     chan struct{}
```

Comment:
Would it be worthwhile to rename `C` to something more descriptive? like `channel` or `discoveryStartedChannel`

Reply:
iirc this is a go convention, but I definitely can make this more clear.